jmxtrans-cluster/src/main/java/com/googlecode/jmxtrans/cluster/zookeeper/ZookeeperClusterService.java

```diff
@@ -0,0 +1,228 @@
+package com.googlecode.jmxtrans.cluster.zookeeper;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Injector;
+import com.googlecode.jmxtrans.cluster.ClusterService;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.configuration.Configuration;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.curator.utils.ZKPaths;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Created by kulcsart on 5/14/2016.
+ *
+ */
+public class ZookeeperClusterService extends Thread implements ClusterService {
+    private static final Logger log = LoggerFactory.getLogger(ZookeeperClusterService.class);
+    private Configuration configuration;
+    private Injector injector;
+
+    private ZookeeperConfig clConfig;
+    private CuratorFramework clClient;
+
+    PathChildrenCache jvmRootCache;
+    PathChildrenCache workerRootCache;
+
+    Map<String, ZookeeperJvmHandler> jvmHandlers= ImmutableMap.of();
+
+
+    @Inject
+    public ZookeeperClusterService(Injector injector, Configuration configuration) {
+        log.info(this.getClass().getName() + " initiated");
+        checkNotNull(this.injector = injector, "Injector cannot be null");
+        checkNotNull(this.configuration = configuration, "Configuration cannnot be null");
+    }
+
+    private void initilaize() {
```

Comment:
Most of this initialization should happen outside of the class (dependency injection and all those nice things...).